client/src/App.js

```diff
@@ -14,15 +15,25 @@ export default class App extends Component {
 
   addToSavedList = movie => {
     const savedList = this.state.savedList;
-    savedList.push(movie);
+    if (savedList.includes(movie)) {
+      return;
+    } else {
+      savedList.push(movie);
+    }
     this.setState({ savedList });
   };
 
   render() {
     return (
       <div>
         <SavedList list={this.state.savedList} />
-        <div>Replace this Div with your Routes</div>
+        <Route exact path="/" component={MovieList} />
+        <Route
+          path="/movies/:id"
+          render={props => {
+            return <Movie {...props} addToSavedList={this.addToSavedList} />;
+          }}
+        />
```

Comment:
Nicely implemented.
Also, you can use implicit returns with arrow functions for your Routes. It's a minor difference and personal preference.
ie: <Route
          path="/movies/:id"
          render={props => (
                 <Movie {...props} addToSavedList={this.addToSavedList} />
          )}
        />